esutil/bulk_indexer.go

```diff
@@ -508,7 +511,9 @@ func (w *worker) flushBuffer(ctx context.Context) error {
 		defer func() { w.bi.config.OnFlushEnd(ctx) }()
 	}
 
-	if w.buf.Len() < 1 {
+	bufLen := w.buf.Len()
+
+	if bufLen < 1 {
 		if w.bi.config.DebugLogger != nil {
```

Comment:
```suggestion
	if bufLen < 1 && w.bi.config.DebugLogger != nil {
```

Reply:
This won't translate to the same code as before because of the `return nil` circuit break below

Reply:
Right